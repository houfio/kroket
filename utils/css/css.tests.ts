import { css } from './index';

afterEach(() => {
  jest.clearAllMocks();
});

it('should return the generated class name', () => {
  expect(css('', false)).toBe('kroket-45h');
});

it('should inject the generated styles', () => {
  css('', false);

  expect(require('@kroket/inject').execute).toHaveBeenCalled();
});

it('should parse data attributes', () => {
  css('[hidden="true"] { opacity: 0; }', false);

  expect(require('@kroket/inject').execute).toMatchSnapshot();
});

it('should parse nested data attributes', () => {
  css('[hidden="true"] { [enabled="false"] { opacity: 0; } }', false);

  expect(require('@kroket/inject').execute).toMatchSnapshot();
});

it('should remember the depth of attributes', () => {
  css('@keyframes test { from { opacity: 1; } to { opacity: 0; } } animation: test;', false);

  expect(require('@kroket/inject').execute).toMatchSnapshot();
});
