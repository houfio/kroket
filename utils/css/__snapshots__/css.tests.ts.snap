// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should parse data attributes 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Array [
        ".kroket-pj0ioc[data-hidden=\\"true\\"]{opacity:0;}",
      ],
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`should parse nested data attributes 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Array [
        ".kroket-17msocp[data-hidden=\\"true\\"][data-enabled=\\"false\\"]{opacity:0;}",
      ],
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`should remember the depth of attributes 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Array [
        ".kroket-e7pk4h{-webkit-animation:test-kroket-e7pk4h;animation:test-kroket-e7pk4h;}",
        "@-webkit-keyframes test-kroket-e7pk4h{from{opacity:1;}to{opacity:0;}}",
        "@keyframes test-kroket-e7pk4h{from{opacity:1;}to{opacity:0;}}",
      ],
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
